Spt renames

ricarkol opened the pull request:
- remove solo5 as a submodule
- point to latest rumprun, rumprun-packages, and gorump
- change a bunch of Makefiles and Dockerfiles to use spt instead of seccomp

Commit:
Removing solo5 from .gitmodules

Commit:
Removed submodule solo5

Commit:
Removing solo5 lib building from Makefile.inc

Signed-off-by: Ricardo Koller <kollerr@us.ibm.com>

Commit:
redis: renames to spt

Signed-off-by: Ricardo Koller <kollerr@us.ibm.com>

Commit:
nginx: renames to spt and enable logging

Signed-off-by: Ricardo Koller <kollerr@us.ibm.com>

Commit:
python3: renames to spt

Signed-off-by: Ricardo Koller <kollerr@us.ibm.com>

Commit:
nodejs: renames to spt

Signed-off-by: Ricardo Koller <kollerr@us.ibm.com>

Commit:
go: renames to spt

Signed-off-by: Ricardo Koller <kollerr@us.ibm.com>

Commit:
Pointing to latest rumprun-packages (with spt renames)

Commit:
Pointing to latest rumprun (with spt renames)

Commit:
Pointing to latest gorump (with spt renames)

Commit:
postmark: renames to spt

Commit:
Removing solo5 target from Makefile.inc

Commit:
Removing solo5 dependency from postmark

Commit:
tests: renames to spt

Commit:
fix travis (use xenial and gcc6)

Commit:
Adding --x-exec-heap to node test

Commit:
Adding source rumprun/.../config-PATH.sh to all bases Makefiles

Commit:
python3: build all first

lumjjb commented on go-base/Dockerfile:
```diff
@@ -2,17 +2,36 @@ FROM ubuntu:16.04
+# rumprun can only be built with gcc-6
+# XXX: Not only that, this version has to match whatever
+# gcc was used to build rumprun in the host.
+RUN apt-get update && \
```
Let's combine this with the previous two commands into 1 `RUN` command so it will just be 1 layer. It'll help us remove the redundant `apt-get update too`

ricarkol replied:
fixed

lumjjb commented on go-base/Dockerfile:
```diff
@@ -2,17 +2,36 @@ FROM ubuntu:16.04
+# needs to be at exactly the same place where it was built.
+# (unless I'm missing something).
+ARG host_rumproot
+ADD rumprun-solo5 ${host_rumproot}
```
Dockerfile best practices: Use `COPY` instead of `ADD`. This is because `ADD` is overloaded.

ricarkol replied:
done

lumjjb commented on go-base/Dockerfile:
```diff
@@ -2,17 +2,36 @@ FROM ubuntu:16.04
 COPY Makefile.goapp /goapp
 
-ENV PATH="/root/unikernels/nabla-base-build/rumprun/rumprun-solo5/bin:${PATH}"
+ENV PATH=${host_rumproot}"/rumprun-solo5/bin:${PATH}"
```
Let's make the quotes consistent? i.e. `ENV PATH="${host_rumproot}/...`

lumjjb commented on go-base/Makefile:
```diff
@@ -39,18 +39,19 @@ gobootstrap: FORCE
-	source $(RUMPRUN)/obj/config-PATH.sh && cd $(GORUMP)/go/src && CGO_ENABLED=0 GOROOT_BOOTSTRAP=$(GOBOOTSTRAP) GOOS=rumprun GOARCH=amd64 ./make.bash && cd $(GOBASE)
+gorump: gobootstrap rumprun FORCE
+	source $(RUMPRUN)/obj/config && source $(RUMPRUN)/obj/config-PATH.sh && \
+		cd $(GORUMP)/go/src && CGO_ENABLED=0 GOROOT_BOOTSTRAP=$(GOBOOTSTRAP) GOOS=rumprun GOARCH=amd64 ./make.bash && cd $(GOBASE)
```
Not really a problem for this PR, but in future for aarch64, we should make `GOARCH` a variable. We can open an issue for this?

ricarkol replied:
Ahh, didn't notice that. Was grepping for x86, will look for amd64 as well. Thanks!

ricarkol replied:
Creating an issue to discuss the proper way of having multiple archs

lumjjb commented on tests/integration/run.bats:
```diff
@@ -23,54 +23,53 @@ function setup() {
 
 @test "test raw node (hello)" {
-	nabla_run --disk="${BATS_TMPDIR}"/dummy ${TOPLEVEL}/node-base/node.nabla
+	nabla_run --x-exec-heap ${TOPLEVEL}/node-base/node.nabla
```
qn: what's the `--x-exec-heap` flag do? and why is it only used in the node case?

ricarkol replied:
It makes the heap executable, it's needed because of Nodejs's JIT compiler. More here: https://github.com/Solo5/solo5/issues/321

ricarkol replied:
Have to add it to `runnc` after pointing to latest there as well.

lumjjb replied:
yikes... I'm curious about some details here.. Wondering if there are any other situations like this. Let's chat about this at some point!

lumjjb reviewed (changes requested):
Some minor nits, if not, LGTM!

Commit:
point to latest rumprun-packages (without CAN sockets support)

Commit:
Fix travis by building and testing less stuff. And adding tests/curl

- add a Makefile rumprun_stamp target so rumprun is not built for every base
- travis only builds node, nginx, and go
- disabled some redis and python tests, as it was taking too much time to build them
and travis times out.
- add test_curl which is used by runnc tests

Commit:
fix makefile for tests

Commit:
Better go-base dockerfile

ricarkol commented:
@lumjjb: went through your comments. Moved the multi-arch issue to #24.

lumjjb reviewed (approved):
LGTM! Thanks for the changes.
